### ISSUE TITLE
Docker login fails with helm chart

### ISSUE THREAD

**Issue by vinayagg** [jfrog/charts#470]
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**: latest

**Which chart**: Artifactory-HA

**What happened**: docker login fails

**What you expected to happen**: docker login to succeed

**How to reproduce it** (as minimally and precisely as possible):
install HA version of artifactory using helm chart
create a docker repo using subdomain method
docker login fails

**Anything else we need to know**:
- Here is a discussion about the issue. 3 lines need to be removed from artifactory config file. They are still present in this chart.
https://stackoverflow.com/questions/46306536/docker-registry-on-artifactory-login-failed
- This is causing our validation to fail on EKS

**Comment by eldada** [comment]
@vinayagg - can you provide some more details on the setup?
Is this using a default deployment? Any customisations? 
If you remove the lines, does it solve your problem?

**Comment by vinayagg** [comment]
Yes it is using default deployment, no customizations at all (in fact our code dynamically pulls helm chart from github during deployment.
Yes, if we remove those three lines, it solves the problem.

**Comment by danielezer** [comment]
@vinayagg I am not able to reproduce this issue. Can you please share your values file and the parameters you provide using --set (if you have any). 
Also, please attach the output of `helm ls`

**Comment by vinayagg** [comment]
@danielezer I am not customizing values files or using any --set values
I was also able to reproduce this with RPM installs of 11.6 version.
Also, the configuration has been fixed in 12.x installs (this is how I found and fixed the issue in 11.6 RPM install). When I found same thing in helm charts, I thought it was a known issue that got fixed in 12.x but not helm charts.
as for `helm ls`, I do not have a cluster running, but I am not installing anything else on the cluster.
If you want me to create a cluster and show you on cluster or give you helm ls etc, i can do that.

**Comment by danielezer** [comment]
When using chart along with the docker subdomain method, you need to either set `ingress.hosts` to the list of DNS names you will access your artifactory installation with, including the DNS names for the docker repo. For example, if you have an artifactory installation under the DNS name `myartifactory.mycompany.com` and docker repositories with the names `docker-repo1` and `docker-repo2`, you will need to set `ingress.hosts` to:
```
- myartifactory.mycompany.com
- docker-repo1.myartifactory.mycompany.com
- docker-repo2.myartifactory.mycompany.com
```
This will cause the nginx configuration to properly infer the repository name from the DNS name you will use to access your repository. 

** It is important to note that you should set `ingress.hosts` for this use case regardless of whether you use ingress or not.
